src/forms/validation-rules.js

```diff
@@ -223,21 +224,26 @@ export function validCollectedVariables(
   }
 
   let codeListPrecision = false;
-  if (expectedVariables.length !== value.length && type === SINGLE_CHOICE) {
+  if (
+    expectedVariables.length !== value.length &&
+    (type === SINGLE_CHOICE || type === PAIRING)
+  ) {
     codeListPrecision = true;
   }
   if (expectedVariables.length !== value.length && type === MULTIPLE_CHOICE) {
     codeListPrecision = true;
   }
   if (
-    (type === SINGLE_CHOICE &&
+    ((type === SINGLE_CHOICE || type === PAIRING) &&
       value[0] &&
       value[0].codeListReference !== expectedVariables[0].codeListReference) ||
-    (type === SINGLE_CHOICE &&
+    ((type === SINGLE_CHOICE || type === PAIRING) &&
       value[0] &&
       value[0].codeListReferenceLabel !==
         expectedVariables[0].codeListReferenceLabel) ||
-    (type === SINGLE_CHOICE && value[0] && codeListPrecision)
+    ((type === SINGLE_CHOICE || type === PAIRING) &&
+      value[0] &&
+      codeListPrecision)
   ) {
```

Comment:
could refactor : `(type === SINGLE_CHOICE || type === PAIRING) &&
      value[0] &&` could be evaluated only once for better readability